js-apps/meep-frontend/src/js/containers/exec/dashboard-container.js

```diff
@@ -133,55 +135,80 @@ const ConfigurationView = props => {
   );
 };
 
+const getUrl = (dashboardName, dashboardOptions) => {
+  var url = '';
+  if (dashboardOptions) {
+    for (var i = 0; i < dashboardOptions.length; i++) {
+      var dashboard = dashboardOptions[i];
+      if (dashboard.label === dashboardName) {
+        url = dashboard.value;
+        url = url.replace(showInExecStr, '');
+        break;
+      }
+    }
+  }
+  return url;
+};
+
 const ViewForName = ({
   scenarioName,
   selectedSource,
   selectedDest,
-  viewName
+  viewName,
+  dashboardOptions
 }) => {
 
-  // Remove '-' from scenario name
-  var scenario = scenarioName.replace(/-/g, '');
-
-  const dashboard = 'http://' + location.hostname + ':30009/d/100/metrics-dashboard?orgId=1';
-  const datasource = '&var-datasource=meep-influxdb';
-  const database = '&var-database=' + scenario;
-  const refreshInterval = '&refresh=1s';
-  const srcApp = '&var-src=' + selectedSource;
-  const destApp = '&var-dest=' + selectedDest;
-  const viewMode = '&kiosk';
-  const theme = '&theme=light';
-  const dashboardUrl = dashboard + datasource + database + refreshInterval + srcApp + destApp + viewMode + theme;
-  
-  switch (viewName) {
-  case METRICS_VIEW:
+  // Handle Network Topology view
+  if (viewName === NET_TOPOLOGY_VIEW) {
     return (
       <div style={{ height: '80vh' }}>
-        <Iframe
-          url={dashboardUrl}
-          id="myId"
-          display="initial"
-          position="relative"
-          allowFullScreen
+        <IDCVis
+          type={TYPE_EXEC}
           width='100%'
           height='100%'
+          onEditElement={() => { }}
         />
       </div>
     );
-  case VIS_VIEW:
+  }
+
+  // Get URL from Monitoring page dashboard options
+  var selectedUrl = getUrl(viewName, DEFAULT_DASHBOARD_OPTIONS);
+  if (selectedUrl === '') {
+    selectedUrl = getUrl(viewName, dashboardOptions);
+  }
+
+  // Add variables if requested
+  if (selectedUrl !== '') {
```

Comment:
It should not be and empty string because the selected view name should be either in the list of default or custom dashboards. However there may be a case where a selected custom dashboard is removed/deleted from the custom list. At this point if we return to the Exec Page the view name will no longer be in the dashboard list. To be safe I would leave the code as is.